clients/go/coreutils/logs/logs.go

```diff
@@ -3,10 +3,52 @@
 package logs
 
 import (
+	"fmt"
 	"github.com/lyft/flyteidl/gen/pb-go/flyteidl/core"
 )
 
+// Deprecated: Please use LogPluginV2
 type LogPlugin interface {
 	// Generates a TaskLog object given necessary computation information
 	GetTaskLog(podName, namespace, containerName, containerID, logName string) (core.TaskLog, error)
 }
+
+type TaskLogInput struct {
+	HostName      string `json:"hostname"`
+	PodName       string `json:"podName"`
+	Namespace     string `json:"namespace"`
+	ContainerName string `json:"containerName"`
+	ContainerID   string `json:"containerId"`
+	LogName       string `json:"logName"`
+}
+
+type LogPluginV2 interface {
+	// Generates a TaskLog object given necessary computation information
+	NewTaskLog(input TaskLogInput) (log core.TaskLog, err error)
+}
+
+// NewCloudwatchLogPlugin creates a TemplateLogPlugin with a preformatted Log links in Cloudwatch for the configured
+// region and group name.
+// Deprecated: Please use NewTemplateLogPlugin instead.
+func NewCloudwatchLogPlugin(region, groupName string) LogPlugin {
```

Comment:
So I wanted to do one more thing along with this. Send back the actual Cloudwatch log-group name and stream name, so that one day we can show the logs in the UI